cmd/service-controller/policy_controller.go

```diff
@@ -35,12 +36,13 @@ func AddStaticPolicyWatcher(pv *client.ClusterPolicyValidator) {
 }
 
 type PolicyController struct {
-	name      string
-	cli       *client.VanClient
-	validator *client.ClusterPolicyValidator
-	informer  cache.SharedIndexInformer
-	queue     workqueue.RateLimitingInterface
-	activeMap map[string]time.Time
+	name          string
+	cli           *client.VanClient
+	validator     *client.ClusterPolicyValidator
+	informer      cache.SharedIndexInformer
+	queue         workqueue.RateLimitingInterface
+	activeMap     map[string]time.Time
+	eventRecorder kube.SkupperEventRecorder
```

Comment:
This struct already has the cli, so there will be a recorder on that (though see comment above) and at this level, which feels redundant and perhaps error prone?

Reply:
It was removed from the VanClient, now the PolicyController is instantiated with a `eventHandler` that comes from the controller.